cocotb/handle.py

```diff
@@ -530,11 +519,69 @@ def _range_iter(self, left, right):
 
     @NonHierarchyObject.value.getter
     def value(self):
-        # need to iterate over the sub-object
-        result = []
-        for x in range(len(self)):
-            result.append(self[x].value)
-        return result
+        """Get value of array by iterating through sub-handles in left-to-right order.
+
+        Arrays with ascending ranges return list with values from sub-handles mapped thusly:
+            Verilog [4:7]       => [4, 5, 6, 7]
+            VHDL    (4 to 7)    => [4, 5, 6, 7]
+
+        Arrays with descending ranges:
+            Verilog [7:4]           => [7, 6, 5, 4]
+            VHDL    (7 downto 4)    => [7, 6, 5, 4]
+
+        Given an HDL array, the returned list contains the values of each sub-handle
+        in the order indicated in the table:
+        +----------------+-----------------------+------------------+
+        | Verilog        | VHDL                  | Python list      |
+        +================+=======================+==================+
+        | ``array[4:7]`` | ``array(4 to 7)``     | ``[4, 5, 6, 7]`` |
```

Comment:
Would this perhaps be clearer as:
```suggestion
        | ``arr[4:7]`` | ``arr(4 to 7)``     | ``[arr[4], arr[5], arr[6], arr[7]]`` |
```